dvc/repo/experiments/show.py

```diff
@@ -142,6 +142,9 @@ def show(
     fetch_running: bool = True,
 ):
 
+    if repo.scm.no_commits:
```

Comment:
exp show is a little different conceptually, but I think it still makes sense to match behavior of the other `diff` commands using `brancher`

Reply:
Should `exp show` (without `--json`) draw an empty table w/the default column headers? or at least output some message saying we are in an empty git repo? I'm not sure that getting no output at all is better than getting the error message in the CLI table case.

Reply:
Appears to be a condition in `TabularData` to not render the table if there is no data (unless markdown flag is passed):
https://github.com/iterative/dvc/blob/ead7724824af068c8a0c0653ce6ae20b77b1b63a/dvc/ui/__init__.py#L235-L236

Not sure what was the original motivation for it, but if removed this would print an empty table

Reply:
In this p.r. `exp show --md` prints:

| Experiment   | Created   |
|--------------|-----------|

Reply:
seems like we should maybe have something like a `show_empty` flag rather than relying on what output format is set

Reply:
Opened https://github.com/iterative/dvc/issues/8179 , can be follow-up